packages/auth/README.md

```diff
@@ -0,0 +1,187 @@
+| user | stores information about the logged-in user |
+
+
+**AuthWall**
+
+```
+<AuthWall 
+  authStore
+  Loading
+  UnauthorizedPage
+  EmailVerificationPage
+  handleTargetUrl
+>
+  <ProtectedComponent />
+</AuthWall>
+```
+
+| Required Props | Description |
+| -: | - |
+| authStore | initialized AuthStore |
+| Loading | component that renders the loading page/state |
+| UnauthorizedPage | component that renders the page for an unauthorized user logging in |
+| EmailVerificationPage | component that renders the page for a user who has not verified their email address |
+
+<br />
+
+| &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Optional Props | Description |
+| -: | - |
+| handleTargetUrl | *handleTargetUrl: ((targetUrl:  string) =>  void) \|  undefined* <br /> runs a callback function, if provided, that will handle the Auth0 `appState` target URL |
+
+### Example Implementation via RootStore and React Context:
+
+First, set up a RootStore (a JavaScript class) that will initialize and house this package's AuthStore. This pattern is particularly useful if you have multiple data stores that you will need to access throughout your application. Read more about this mobX pattern here: [Defining data stores](https://mobx.js.org/defining-data-stores.html)
+
+*RootStore.ts*
+```
+import { AuthStore } from "@recidiviz/auth";
+
+const authSettings = {
+  domain: <AUTH0 DOMAIN>,
+  client_id: <AUTH0 CLIENT_ID>,
+  redirect_uri: <AUTH0 REDIRECT_URI>,
+  audience: <AUTH0 AUDIENCE>,
+  useRefreshTokens: <true | false>,
+};
+
+class RootStore {
+  authStore: AuthStore;
+  
+  constructor() {
+	this.authStore = new AuthStore({ authSettings });
+  }
+}
+
+export default new RootStore();
+export type { RootStore };
+```
+
+Next, set up a [React context provider](https://reactjs.org/docs/context.html) that will allow you to utilize the `authStore` throughout your application by importing `useStore` and destructuring the `authStore` --- e.g. `const { authStore } = useStore()`.
+
+*StoreProvider.tsx*
+```
+import React, { useContext } from "react";
+import rootStore from "./RootStore";
+
+const StoreContext = React.createContext<typeof rootStore | undefined>(undefined);
+
+export const StoreProvider: React.FC<React.ReactNode> = ({
+  children
+}): React.ReactElement => (
+  <StoreContext.Provider value={rootStore}>{children}</StoreContext.Provider>
+);
+
+export function useStore(): typeof rootStore {
+  const context = useContext(StoreContext);
+
+  if (context === undefined) {
+    throw new Error("useStore must be used within a StoreProvider");
+  }
+
+  return context;
+}
+```
+
+  Import `AuthWall`, `rootStore`, `StoreProvider`, required `AuthWall` prop components and components you want protected behind an auth wall --- and you are all set!
+
+*App.tsx*
+```
+import React from "react";
+import { BrowserRouter } from "react-router-dom";
+
+import { AuthWall } from "@recidiviz/auth";
+import { rootStore, StoreProvider } from "./stores";
```

Comment:
I was thinking about this adapted pattern... This `rootStore` - it's setup exporting an initialized RootStore (`export default new RootStore();`). The `rootStore` that we are feeding into the `StoreProvider` as well as here in the `AuthWall` - is it the same rootStore or separate rootStores? I worry it's the latter.

Reply:
it should be the same instance because that is the only thing that is exported by `RootStore.ts`. 

However, this was not your question but I don't think we even need this level of detail in the example, since the package itself does not care how the application organizes its datastores or for that matter whether it even has any .. I would just say that this thing is a datastore, link out to the Mobx docs about datastores, and gloss over all the stuff about the root store and the context provider. 

While we have used this singleton pattern for the rootstore before, in retrospect I think it is not the best and we would be better off not blessing it as the recommended default

Reply:
This answered my question and more! Thanks, Ian. I agree and will trim the excess details!